tensorflow/python/ops/image_grad.py

```diff
@@ -153,3 +155,207 @@ def _CropAndResizeGrad(op, grad):
       grad, op.inputs[0], op.inputs[1], op.inputs[2])
 
   return [grad0, grad1, None, None]
+
+
+def _CustomReciprocal(x, my_eps=1e-10):
+  """
+  Performs reciprocal with an eps added to the input. This is to avoid
+  inversion errors or divide by zeros or NaNs.
+  Inputs:
+    x -> input tensor to be reciprocat-ed
+    my_eps -> custom machine precision epsilon
+  Returns:
+    x_reciprocal -> reciprocal of x added with my_eps
+  """
+  return reciprocal(x + my_eps)
```

Comment:
This is not an ideal implementation, as it can turn -eps into 0, for example.

Instead, use math_ops.div_no_nan or tf.where to make this work correctly.